python-threatexchange/threatexchange/cli/hash_cmd.py

```diff
@@ -101,13 +101,17 @@ def execute(self, settings: CLISettings) -> None:
         str_hashers = [s for s in all_signal_types if issubclass(s, TextHasher)]
 
         for inp in self.input_generator:
-            hash_fn = lambda s, t: s.hash_from_file(t)
-            signal_types: t.List[t.Any] = file_hashers
-            if isinstance(inp, str):
-                hash_fn = lambda s, t: s.hash_from_str(t)
-                signal_types = str_hashers
-
-            for signal_type in signal_types:
-                hash_str = hash_fn(signal_type, inp)
-                if hash_str:
-                    print(signal_type.get_name(), hash_str)
+            try:
+                hash_fn = lambda s, t: s.hash_from_file(t)
+                signal_types: t.List[t.Any] = file_hashers
+                if isinstance(inp, str):
+                    hash_fn = lambda s, t: s.hash_from_str(t)
+                    signal_types = str_hashers
+                for signal_type in signal_types:
+                    hash_str = hash_fn(signal_type, inp)
+                    if hash_str:
+                        print(signal_type.get_name(), hash_str)
+            except FileNotFoundError:
+                self.stderr(
+                    f"The file {inp} doesn't exist or the file path is incorrect"
+                )
```

Comment:
(blocking): You want to make this try: catch narrower to the statements you expect to throw this exception. In this case it's L111